Include Kotlin DSL samples for chapter dependency_management_for_java_projects

jnizet opened the pull request:
Important notes:
- the chapter repository_types is also done, because it shared examples with dependency_management_for_java_projects
- the chapter dependency_types is partially done, because it shared examples with dependency_management_for_java_projects

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

StefMa commented on subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,305 @@
+// tag::maven-ivy-repository-no-auth[]
+repositories {
+    maven {
+        setUrl("http://repo.mycompany.com/maven2")
```
Use the overload `maven(url=...)`

jnizet replied:
I didn't noticed this overload existed. But I also tried to stay as close as the groovy script as possible. Let's see what the team members prefer here.

StefMa commented on subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,305 @@
+    }
+
+    ivy {
+        setUrl("http://repo.mycompany.com/repo")
```
Same like above... 👆

StefMa commented on subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,305 @@
+repositories {
+    maven {
+        setUrl("s3://myCompanyBucket/maven2")
+        credentials(AwsCredentials::class.java) {
```
Is using `.java` still needed? 🤔 
I thought with 4.10-rcX it is possible to use the Kotlin classes?

jnizet replied:
Indeed, `AwsCredentials::class` works fine. Unfortunately, this is yet another case where The build is successful, but IntelliJ marks the code as red (`Type mismatch. Required: Class<???..???> Found KClass<AwsCredentials>`).
I'll leave this as is for now, until a core member notices the issue and advises.

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/externalDependencies/groovy/build.gradle:
```diff
@@ -7,6 +7,22 @@ configurations {
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
```
:x: this header is out of place

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/externalDependencies/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,40 @@
+        setTransitive(true)
+    }
+    runtime(group = "org.hibernate", name = "hibernate", version = "3.0.5") {
+        setTransitive(true)
```
:x: this can be replaced by the simpler `isTransitive = true` here and in other places in this PR

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,303 @@
+repositories {
+    maven {
+        url = uri("s3://myCompanyBucket/maven2")
+        credentials(AwsCredentials::class.java) {
```
:x: please use `credentials(AwsCredentials::class) {` instead, and in other places in this PR

IntelliJ still wrongly highlights this as an error but it's already fixed in Kotlin 1.2.70 development builds, see https://github.com/gradle/kotlin-dsl/issues/1064

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/externalDependencies/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,40 @@
+// end::define-dependency[]
+
+// tag::use-configuration[]
+tasks.create("listJars") {
```
:x: `task("listJars") {`

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/defineRepository/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,303 @@
+}
+// end::authenticated-ivy-repo[]
+
+tasks.create("lookup") {
```
:x: `task("lookup") {`

eskatos commented on subprojects/docs/src/samples/userguide/artifacts/externalDependencies/kotlin/build.gradle.kts:
```diff
@@ -0,0 +1,40 @@
+// tag::file-dependencies[]
+dependencies {
+    runtime(files("libs/a.jar", "libs/b.jar"))
+    runtime(fileTree("dir" to "libs", "include" to "*.jar"))
```
❓ this could be changed to the more explicit `runtime(fileTree("libs") { include("*.jar") })`

jnizet replied:
I guess so, but I chose to stay close to the groovy sample, which uses `runtime fileTree(dir: 'libs', include: '*.jar')`.

I'll change both samples to keep the consistency.

jnizet commented:
changes done @eskatos

jnizet commented:
@eskatos I fixed the test.

I wouldn't dare deleting a test without asking first, but I think this test should be deleted. 

All it does is to execute the `lookup` task on the build and check it doesn't fail. So I modified the exemplar configuration for this sample in order to also execute the `lookup` task rather than the `tasks` task, and the exemplar test should thus now do the same thing as this integration test, making it redundant.
